presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package me.androidbox.presentation.agenda.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import me.androidbox.presentation.agenda.screen.AgendaScreenEvent
+import me.androidbox.presentation.agenda.screen.AgendaScreenState
+import javax.inject.Inject
+
+@HiltViewModel
+class AgendaViewModel @Inject constructor(
+    private val preferenceRepository: PreferenceRepository
+) : ViewModel() {
+
+    private val _agendaScreenState = MutableStateFlow(AgendaScreenState())
+    val agendaScreenState = _agendaScreenState.asStateFlow()
+
+    init {
+        getAuthenticatedUser()
+    }
+
+    fun onAgendaScreenEvent(agendaScreenEvent: AgendaScreenEvent) {
+        when(agendaScreenEvent) {
+            is AgendaScreenEvent.OnDateChanged -> {
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        displayMonth = agendaScreenEvent.date.month.toString()
+                    )
+                }
+            }
+        }
+    }
+
+    private fun getAuthenticatedUser() {
+        viewModelScope.launch {
+            preferenceRepository.retrieveCurrentUserOrNull()?.let { authenticatedUser ->
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        usersInitials = authenticatedUser.fullName.extractInitials()
+                    )
+                }
+            }
+        }
+    }
+
+    private fun String.extractInitials() : String {
+        val listOfFullName = this.split(" ")
+        var initials = ""
+
+        listOfFullName.forEach { name ->
+            initials += name.first()
+        }
+
+        return initials.uppercase()
+    }
+}
```

Comment:
@philipplackner @MKiperszmid This is a extension method that extracts the initials from the full name. An edge case could be in the user has a name that is more than 2 names. i.e. a middle name.
Should this be in a utility package or is it ok to have it here, if this is the only place we extract the initials.

Reply:
Opinionated answer:

I like to do the following
If I'm using it only once/twice, I have a Util class

If I use it in many places, then I have an extension function



Btw, this is a great method to unit test ;)

Reply:
Either way this is business logic and should go in domain therefore. Middle names are accepted and need to be handled here. In that case the initials would still be first name and last name's initials. 
Don't remember if you use use cases for your app, but if so, it should go in a use case. If not, it can be any type of util class